### ISSUE TITLE
Model Converge Problem

### ISSUE THREAD

**Issue by khawar512** [google-research/nested-transformer#5]
I am training on a medium-scale dataset that consists of 100,000 images. The learning rate and weight decay as the same as your config but still not working. Any opinion?

Regards,
Khawar Islam

**Comment by zizhaozhang** [comment]
What do you mean by not working, totally disconverge? Or performance not good. It will be useful if you are provide more information.  
Two suggestions:
- It is better to do data diagnosis first.
- Train a standard ResNet on your dataset, to see whether it is data issue or model issue.

**Comment by zizhaozhang** [comment]
Did you train these methods from scratch or finetune with their pre-trained checkpoints. It will be important sometimes. 
Our scripts currently only train from scratch, but it can be easy for finetuning using our pre-trained models.

**Comment by khawar512** [comment]
Did you train these methods from scratch or finetune with their pre-trained checkpoints. It will be important sometimes.
I am training from scratch

Our scripts currently only train from scratch, but it can be easy for finetuning using our pre-trained models.
scratch

**Comment by zizhaozhang** [comment]
Hi, from our experiments, we do not our methods have convergence issue. it will be great if you can provide more training detailed info so I can help, e.g. what is the others method you train and what is the setup (scripts), what is the number of devices. Otherwise, it is hard to diagnose.

**Comment by Euruson** [comment]
The network is kind of "sensitive". I used AdamW with learning rate decay and found it crashed when adjusting the learning rate. Note that I used the implementation of PyTorch in timm.

![image](https://user-images.githubusercontent.com/27568823/142134512-66729ccf-cd31-47b7-987e-c4ad6461fd38.png)

**Comment by Freder-chen** [comment]
> The network is kind of "sensitive". I used AdamW with learning rate decay and found it crashed when adjusting the learning rate. Note that I used the implementation of PyTorch in timm.
> 
> ![image](https://user-images.githubusercontent.com/27568823/142134512-66729ccf-cd31-47b7-987e-c4ad6461fd38.png)

From my training, this should be a rare occurrence. And it is recommended to use gradient clipping.

**Comment by abdohelmy** [comment]
Hi, From my experience with the architecture. It is very sensitive to warm-up epochs. When i used the implementation from timm with the warm up schedule of PyTorch-lightning it was diverging. But when i followed their warm up implementation it worked fine. The problem also happened with me once after when i did not use any augmentation by mistake.
